File: gaia-sdk-javascript/src/graphql/response/type/Evaluation.ts

import {BuildInEvaluation} from "./BuildInEvaluation";
import {SkillEvaluation} from "./SkillEvaluation";

import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../../request/enumeration/Order";
import {OrderByField} from "../../request/enumeration/OrderByField";
import {EdgeOrderByField} from "../../request/enumeration/EdgeOrderByField";

export interface Evaluation {
    skill?:SkillEvaluation, 
    buildIn?:BuildInEvaluation
}

File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/request/type/Evaluation.kt
package gaia.sdk.request.type

import gaia.sdk.client.Type
import gaia.sdk.request.intf.*
import gaia.sdk.client.Input
import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.request.input.*
import gaia.sdk.request.enumeration.*

class Evaluation: Type() {

    fun skill(config: SkillEvaluation.() -> Unit) = 
        add { "skill{ " + SkillEvaluation().apply(config).render(it) + "}"}


    fun buildIn(config: BuildInEvaluation.() -> Unit) = 
        add { "buildIn{ " + BuildInEvaluation().apply(config).render(it) + "}"}

}

